plasmapy/plasma/grids.py

```diff
@@ -734,3 +888,88 @@ def _make_mesh(self, start, stop, num, **kwargs):
         arr0, arr1, arr2 = np.meshgrid(ax0, ax1, ax2, indexing="ij")
 
         return arr0, arr1, arr2
+
+
+def example_grid(name, L=1 * u.cm, num=100):
```

Comment:
I'm having a hard time wrapping my head around this function.  It feels more like a "field generator" than an example grid.  And, as an example grid it's opaque since it does not demonstrate how to build a grid.  I'd rather see this function postponed to the fields PR and have a notebook that builds a grid for an electrostatic sphere than one the uses `example_grid()` to get the electrostatic sphere grid and plot it.  That approach would be a lot more informative for the user.  Now, when the field functionality is added down the road we could then update the notebook to compare constructing the grid to the built-in functionality (not absolutely necessary though).

Reply:
Could be good to rename it, say, to example field generator, but if it's there already, achieves what it's supposed to add is going to be helpful later I don't see any value in postponing it. That just creates administrative overhead, I think.

Reply:
I agree with Dominik: I'm fine with renaming this function but it already accomplishes everything I was previously doing with `field.py`, so I think having another type of object is redundant at this point. 

I wouldn't call it `field_generator`, though, because it could just as easily construct grids with more than just fields on them, eg. density profiles for testing interferometry synthetic diagnostics. Maybe `template_grids` or `test_grids`?

Reply:
I'm still having a hard time with this.  I don't see when the current form would be used.  I think a more usefully form would be a `populate_grid()` where a user would request a basic shape/type/template (i.e. hard sphere, gaussian sphere, cylinder, etc.) and the variable type (i.e. density, b-field, e-field, etc.).  I'd see this working something like...

```python
grid = grids.CartesianGrid(x, y, z)

# add a hard sphere of density
grid = populate_grid(grid, template="hard sphere", quantity_type="density", quantity_units=u.cm**-3)

# add a hard cylinder of magnetic field along the x-axis
grid = populate_grid(
    grid, 
    template="hard cylinder",
    quantity_type="magnetic field",
    quantity_units=u.T,
    align_on=ax,
)
```

This is just a concept off the top of my head.  I think dealing with 1D,  2D, and 3D data in one function `populate_gird()` might be a bit difficult.

----

With that said, I think your "Example Grids" notebook would be more useful to a user if you, say, went through the process of constructing an electrostatic Gaussian sphere first and then was like, "Ah ah, we do the hard work for you and you can use `example_grids()` instead."

Reply:
So the only use-case I had in mind was for example notebooks, and I see your point that manually creating these grids in those notebooks is more instructive. So, I have removed the `example_grids` function and corresponding notebook from this PR. I will create these in the notebook examples for different synthetic diagnostics as they come up. 

I like the idea of a `populate_grid` function, but that sounds like a considerable project. Let's leave it for another PR - we don't need it for the current synthetic diagnostics project.

Reply:
I think that's a good route to take, and thank you for putting up with my hardheadedness on this.  Your original concept behind `example_grids` is a very good one and, if we generalize it little more, then I think it will be a very useful tool.